runtimes/node-18.0/server.js

```diff
@@ -1,85 +1,239 @@
+                response.body = body;
+                return this;
+            },
+
+            setStatusCode: function (statusCode) {
+                response.statusCode = statusCode;
+                return this;
+            },
+
+            setContentType: function (contentType) {
+                response.contentType = contentType;
+                return this;
+            },
+
+            setCookies: function (cookies) {
+                response.cookies = cookies;
+                return this;
+            },
+
+            setHeaders: function (headers) {
+                response.headers = headers;
+                return this;
+            },
+
+            addHeader: function (key, value) {
+                response.headers[key] = value;
+                return this;
+            },
+
+            addCookie: function (name, value, expire, maxage, path, domain, secure, httponly, samesite) {
+                response.cookies[name] = { value, expire, maxage, path, domain, secure, httponly, samesite };
+            },
+
+            send: function (body, statusCode, headers) {
+                if (body !== undefined) { response.body = body; }
+                if (statusCode !== undefined) { response.statusCode = statusCode; }
+                if (headers !== undefined) { response.headers = headers; }
+
+                const resHeaders = { ...response.headers };
+
+                if (response.contentType) {
+                    resHeaders['content-type'] = response.contentType;
+                }
+
+                if (response.cookies && Object.keys(response.cookies).length > 0) {
+                    resHeaders['set-cookie'] = Object.keys(response.cookies).map((name) => {
+                        const cookie = response.cookies[name];
+
+                        const attributes = [];
+                        attributes.push(`${name}=${cookie.value}`);
+
+                        if (cookie.domain) {
+                            attributes.push(`Domain=${cookie.domain}`);
+                        }
+                        if (cookie.expire) {
+                            attributes.push(`Expire=${cookie.expire}`);
+                        }
+                        if (cookie.maxage) {
+                            attributes.push(`Max-Age=${cookie.maxage}`);
+                        }
+                        if (cookie.path) {
+                            attributes.push(`Path=${cookie.path}`);
+                        }
+                        if (cookie.samesite) {
+                            attributes.push(`SameSite=${cookie.samesite}`);
+                        }
+                        if (cookie.secure) {
+                            attributes.push(`Secure`);
+                        }
+                        if (cookie.httponly) {
+                            attributes.push(`HttpOnly`);
+                        }
+
+                        return attributes.join('; ');
+                    }).join('; ');
```

Comment:
Is this nessessary for first iteration? Devs can still set cookies on their own with headers. This is making it simpler, but more complex on the Open Runtimes side. I would prefer having this as improvement, alongside better body parsing (XML,formData,..)